operators/constellation-node-operator/internal/cloud/aws/client/nodeimage.go

```diff
@@ -217,3 +218,10 @@ func (c *Client) DeleteNode(ctx context.Context, providerID string) error {
 func toPtr[T any](v T) *T {
 	return &v
 }
+
+func isInstanceNotFoundError(err error) bool {
+	if err == nil {
+		return false
+	}
+	return strings.Contains(err.Error(), "Instance Id not found")
+}
```

Comment:
```suggestion
func isInstanceNotFoundError(err error) bool {
	var respErr *awshttp.ResponseError // using "awshttp" for package github.com/aws/aws-sdk-go-v2/aws/transport/http
	return errors.As(err, &respErr) && respErr.HTTPStatusCode() == http.StatusNotFound
}

```
From https://github.com/aws/aws-sdk-go-v2/blob/main/CHANGELOG.md#error-handling
Did not tests this, but if it works it would be preferable  over string comparisons.

Reply:
That would be preferable indeed, but AWS does not return 404s in that case, but 400s. I meant to also link to the logs but forgot - fixed that now (see above). I would not want to infer nonexistence from a 400, because that might hide other failure conditions, so I'm back at string comparison :/ 

```xml
<ErrorResponse xmlns="http://autoscaling.amazonaws.com/doc/2011-01-01/">
  <Error>
    <Type>Sender</Type>
    <Code>ValidationError</Code>
    <Message>Instance Id not found - No managed instance found for instance ID: i-061c63c5eb45f0416</Message>
  </Error>
  <RequestId>702f44ea-6d55-480b-8dfa-3da9e62f1634</RequestId>
</ErrorResponse>
```